### PR TITLE
Add functions.c file, printf.c file, & main.h header

### PR DESCRIPTION
**Pull request by MoonTheRipper**
# Added The C files for Task 0 in the project. 
- awaiting approval to est on the intranet
- Please be active on this.

### CODE REVIEW

File: functions.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			base = 16;
+		count = _putdig((va_arg(arg, unsigned int)), base);
+	}
+	else
+	{
+		count += write(1, &formatSpecifier, 1);
+	}
+	return (count);
+}
+
```
Comment (MoonTheRipper) [inline functions.c:55]: The Putchar function works well

File: functions.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+int _putchar(int c)
+{
+	return (write(1, &c, 1));
+}
+
+/**
+ * _putstr - formats a string
+ * @str: pointer to string
+ * Return: bytes used in stdout
+ */
```
Comment (MoonTheRipper) [inline functions.c:70]: the putstr works well

File: functions.c [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ * @formatSpecifier: this is the s, c, d ...
+ * @arg: This is the list of arguments passed.
+ * Return: the value fom the called function.
+ * Description: The function choosed a function
+ *				that formats output depending on the specifier
+ *				I chooses using "ar_arg" whether to pas a type
+ *				to a specific funcion as per the specifier.
+ *				Not theat During arg passing to the {...}
+ *				types are promoted eg char becomes an Int.
+ */
```
Comment (MoonTheRipper) [inline functions.c:14]: the format selector ```printer``` works well

File: functions.c (hunk 2 above)
Comment (larinegit) [reply functions.c:70]: let me review it

**Review by larinegit — APPROVED** [review]

**Pull request merged**